src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -519,15 +519,25 @@ private boolean checkForShovel()
         return needsShovel;
     }
 
-    private boolean checkForTool(@NotNull final String tool)
+     /**
+     * Ensures that we have a tool available.
+     *
+     * @param tool tool required for block
+     * @return true if we have a tool
+     */
+    private boolean checkForTool(@NotNull String tool)
     {
         final boolean needsTool = !InventoryFunctions
                                      .matchFirstInInventory(
                                        worker.getInventoryCitizen(),
                                        stack -> Utils.isTool(stack, tool),
                                        InventoryFunctions::doNothing
                                      );
-        if (!needsTool)
+
+        final boolean isUsable = hasToolLevel(tool);
+        final int hutLevel = worker.getWorkBuilding().getBuildingLevel();
+
+        if (!needsTool && isUsable)
```

Comment:
the needsTool doesn't take into consideration the level, so, a worker with level 0 hut "doesn't need a shovel" if he already has a diamond shovel in his inventory, but since it's a diamond one, he can't use it, therefore we need those 2 to be verified